conjur_api/http/api.py

```diff
@@ -156,6 +156,35 @@ async def resources_list(self, list_constraints: dict = None) -> dict:
         # ?tocpath=Developer%7CREST%C2%A0APIs%7C_____17
         return resources
 
+    async def check_privilege(self, kind: str, resource_id: str, privilege: str, role_id: str) -> bool:
+        """
+        This method is used to check for a privilege on a resource.
+        """
+        params = {
+            'account': self._account,
+            'kind': kind,
+            'identifier': resource_id,
+            'privilege': privilege,
+            'role': role_id if role_id else "",
+        }
+        params.update(self._default_params)
+
+        try:
+            await invoke_endpoint(HttpVerb.HEAD, ConjurEndpoint.PRIVILEGE,
+                                  params,
+                                  api_token=await self.api_token,
+                                  ssl_verification_metadata=self.ssl_verification_data)
+        except HttpStatusError as err:
+            if err.status == 204:
+                return True
+            if err.status == 401:
+                return False
+            if err.status == 404:
+                return False
```

Comment:
If an error occurs with a different code, we would want to `raise` it here. The way it is currently it looks like this function would return `True` if for example a 500 server error occurred since it doesn't hit any of of the conditionals and True is the default return value. [See here.](https://github.com/cyberark/conjur-api-python/blob/18a6606f521b1cbc53a78bfd0c3d604931ea0228/conjur_api/http/api.py#L205)

Reply:
added the 'raise' in most recent commit to handle other error codes

Reply:
added the fix in my most recent commit